Mathlib/Data/Finset/Lattice.lean

```diff
@@ -1181,27 +1181,33 @@ theorem mem_sup {α β} [DecidableEq β] {s : Finset α} {f : α → Multiset β
 end Multiset
 
 namespace Finset
+variable [DecidableEq α] {s : Finset ι} {f : ι → Finset α} {a : α}
 
-theorem mem_sup {α β} [DecidableEq β] {s : Finset α} {f : α → Finset β} {x : β} :
-    x ∈ s.sup f ↔ ∃ v ∈ s, x ∈ f v := by
-  change _ ↔ ∃ v ∈ s, x ∈ (f v).val
-  rw [← Multiset.mem_sup, ← Multiset.mem_toFinset, sup_toFinset]
-  simp_rw [val_toFinset]
+set_option linter.docPrime false in
+@[simp] lemma mem_sup' (hs) : a ∈ s.sup' hs f ↔ ∃ i ∈ s, a ∈ f i := by
+  induction' hs using Nonempty.cons_induction <;> simp [*]
+
+set_option linter.docPrime false in
+@[simp] lemma mem_inf' (hs) : a ∈ s.inf' hs f ↔ ∀ i ∈ s, a ∈ f i := by
```

Comment:
I'm missing `mem_inf`, but that seems to be a pattern in this file. Arguably somewhat out of scope for this PR.

Reply:
`mem_inf` requires `Fintype` (and that's the only one out of the four with this property)

Reply:
I think we still ought to have `mem_inf`, with the `Fintype` instance (and probably a comment saying the fintype instance is necessary), but that can be in a different PR if you prefer.

Reply:
Sorry, I was unclear: `Finset.mem_inf` already exists in a file with access to the `Fintype` instance